File: component-acgnews/src/main/java/com/rabtman/acgnews/di/module/ZeroFiveNewsItemModule.java
package com.rabtman.acgnews.di.module;

import com.rabtman.acgnews.mvp.contract.ZeroFiveNewsContract;
import com.rabtman.acgnews.mvp.model.ZeroFiveNewsModel;
import com.rabtman.common.di.scope.FragmentScope;
import dagger.Module;
import dagger.Provides;

/**
 * @author Rabtman
 */
@Module
public class ZeroFiveNewsItemModule {

  private ZeroFiveNewsContract.View view;

  public ZeroFiveNewsItemModule(ZeroFiveNewsContract.View view) {
    this.view = view;
  }

  @FragmentScope
  @Provides
  ZeroFiveNewsContract.View providerZeroFiveNewsView() {
    return this.view;
  }

  @FragmentScope
  @Provides
  ZeroFiveNewsContract.Model providerZeroFiveNewsModel(ZeroFiveNewsModel model) {
    return model;
  }
}

File: component-acgnews/src/main/java/com/rabtman/acgnews/di/module/ZeroFiveNewsItemModule.kt
package com.rabtman.acgnews.di.module

import com.rabtman.acgnews.mvp.contract.ZeroFiveNewsContract
import com.rabtman.acgnews.mvp.model.ZeroFiveNewsModel
import com.rabtman.common.di.scope.FragmentScope
import dagger.Module
import dagger.Provides

/**
 * @author Rabtman
 */
@Module
class ZeroFiveNewsItemModule(private val view: ZeroFiveNewsContract.View) {
    @FragmentScope
    @Provides
    fun providerZeroFiveNewsView(): ZeroFiveNewsContract.View {
        return view
    }

    @FragmentScope
    @Provides
    fun providerZeroFiveNewsModel(model: ZeroFiveNewsModel): ZeroFiveNewsContract.Model {
        return model
    }

}